### DISCUSSION TITLE
RTKQ Accumulate results and fetch only for missing parameters

### DISCUSSION THREAD

**Discussion by dumaron** [reduxjs/redux-toolkit#3896] [Q&A] [answered]
Hello everyone,
I'm facing an issue with modelling the interaction with an API in RTKQ, and I'd like to have your thoughts on the topic.

To make things simpler, let's say the API consists of a single POST endpoint where it's possible to retrieve elements by id:
```js
fetch(
   '/my-api', 
   { body: { ids: [ 1, 2, 3 ] }}
)
```

Which returns something like:
```js
{
   counts: {
      1: 'data1',
      2: 'data2',
      3: 'data3',
   }
}
```

Now, I think I could easily model that with a `fetchBaseQuery`. However, here comes the weird interaction part: my goal is to:
1. fetch only the data for ids I have no data for;
2. retrieve all fetched ids at once

Having some sort of hooks that returns me a structure like `{ 1: 'data1', 2: 'data2`, ... }` I could technically satisfy both needs given that I could know the ids already fetched and have all the data. So the next call could be:
```js
fetch(
   '/my-api', 
   { body: { ids: [ 4, 5 ] }}
)
```
and have the final hook to return all the fetched ids so far (so also 1, 2, 3).

I don't need the API slice to find out what ids are missing, I can do it myself with a custom hook, but I need to have a way to accumulate all retrieved data and get all at once with a hook.

Do you think it's possible to implement this interaction with RTKQ?

**Comment by markerikson** [comment, marked as answer]
That doesn't sound like something that RTKQ is well suited for.  It's designed to be a "document cache", storing exactly what was sent back by the server (possibly modified by `transformResponse`).  It sounds like you're asking for something more along the lines of a "normalized cache", like Apollo or Urql

- https://redux-toolkit.js.org/rtk-query/usage/cache-behavior#no-normalized-or-de-duplicated-cache
